src/common/wui_api.c

```diff
@@ -7,5 +7,104 @@
  */
 #include "wui_api.h"
 
-void get_eth_params(wui_eth_params_s eth_params) {
+#include "ini_handler.h"
+#include "eeprom.h"
+#include "string.h"
+
+#define LAN_MSK_TYPE      2 //EEPROM mask for user-defined settings (Switch between dhcp and static)
+#define IP4_TYPE_FLAG_POS 1
+
+#define IS_LAN_STATIC(flg) (flg & LAN_MSK_TYPE)
+#define IS_LAN_DHCP(flg)   (!IS_LAN_STATIC(flg))
+
+static int ini_handler_func(void *user, const char *section, const char *name, const char *value) {
+
+    ETH_Config_t *tmp_config = (ETH_Config_t *)user;
+
+#define MATCH(s, n) strcmp(section, s) == 0 && strcmp(name, n) == 0
+
+    if (MATCH("lan_ip4", "type")) {
+        if (strncmp(value, "DHCP", 4) == 0 || strncmp(value, "dhcp", 4) == 0) {
+            tmp_config->lan_flag &= ~LAN_EEFLG_TYPE;
```

Comment:
This should be readable, under some macro maybe.. (CHANGE_TO_DHCP(tmp_config->lan_flag))